ninja-core/src/main/java/ninja/params/ParamParsers.java

```diff
@@ -53,61 +60,95 @@
                     .put(short.class, new PrimitiveShortParamParser())
                     .put(Character.class, new CharacterParamParser())
                     .put(char.class, new PrimitiveCharacterParamParser())
+                    .put(Date.class, new DateParamParser())
                     .build();
 
-    private static final Map<Class<? extends Enum<?>>, ParamParser<?>> ENUM_PARSERS = Maps.newHashMap();
-
-    public static ParamParser<?> getParamParser(Class<?> targetType) {
-
+    private static final Map<Class<?>, Class<? extends ParamParser<?>>> PARAM_PARSERS_CLASSES = Maps.newHashMap();
+    
+    /**
+     * ALlows to register a new parser for given Class parameters.
+     */
+    public static <T> void registerParamParser(final Class<T> paramClass, final Class<? extends ParamParser<T>> parserClass) {
```

Comment:
I'd like to have a big warning sign that registerParamParser must only be used at startup of Ninja. All other usages will not work in a multi threaded context with PARAM_PARSERS_CLASSES's HashMap. The same should be added for unregisterParamParsers.

Maybe that is another sign to rewrite that class to a non static class (Singleton). (I guess not part of this PR, just saying).